pegen/pegen.c

```diff
@@ -693,3 +706,33 @@ store_name(Parser *p, expr_ty load_name)
                     Store,
                     EXTRA_EXPR(load_name, load_name));
 }
+
+expr_ty
+del_name(Parser *p, expr_ty load_name)
+{
+    return _Py_Name(load_name->v.Name.id,
+                    Del,
+                    EXTRA_EXPR(load_name, load_name));
+}
+
+asdl_seq *
+map_targets_to_del_names(Parser *p, asdl_seq *seq)
+{
+    int len = asdl_seq_LEN(seq);
+    asdl_seq *new_seq = _Py_asdl_seq_new(len, p->arena);
+    for (int i = 0; i < len; i++) {
+        expr_ty e = asdl_seq_GET(seq, i);
+        if (e->kind == Name_kind) {
+            asdl_seq_SET(new_seq, i, del_name(p, e));
+        } else if (e->kind == Tuple_kind) {
+            asdl_seq_SET(new_seq, i, _Py_Tuple(map_targets_to_del_names(p, e->v.Tuple.elts),
+                                               Del,
+                                               EXTRA_EXPR(e, e)));
+        } else if (e->kind == List_kind) {
+            asdl_seq_SET(new_seq, i, _Py_List(map_targets_to_del_names(p, e->v.List.elts),
+                                              Del,
+                                              EXTRA_EXPR(e, e)));
+        }
```

Comment:
There are other possible kinds here -- a target may also take the form `a[i]` or `o.a`. Since those aren't yet supported in the grammar (they have no actions) it doesn't matter, but I'd at least add a comment, or if it doesn't break the tests an assert.